src/gui/gui_nation_picker.hpp

```diff
@@ -200,10 +200,24 @@ class saves_window : public window_element_base {
 
 class pick_nation_button : public button_element_base {
 public:
+	void on_update(sys::state& state) noexcept override {
+		auto n = retrieve<dcon::nation_id>(state, parent);
+		if(state.network_mode == sys::network_mode::single_player) {
+			disabled = false;
+		} else {
+			// Prevent (via UI) the player from selecting a nation already selected by someone
+			disabled = !command::can_switch_nation(state, state.local_player_nation, state.world.nation_get_identity_from_identity_holder(n));
+		}
+	}
+
 	void button_action(sys::state& state) noexcept override {
 		auto n = retrieve<dcon::nation_id>(state, parent);
-		state.local_player_nation = n;
-		state.ui_state.nation_picker->impl_on_update(state);
+		if(state.network_mode == sys::network_mode::single_player) {
+			state.local_player_nation = n;
+			state.ui_state.nation_picker->impl_on_update(state);
+		} else {
+			command::switch_nation(state, state.local_player_nation, state.world.nation_get_identity_from_identity_holder(n));
```

Comment:
Have you accounted for the fact that the player don't have a valid nation id when the game opens? This command will already read as being sent from REB or whatever, and the normal version of this command IIRC sets the player controlled flag, which is not appropriate yet